bin/github-review.sh

```diff
@@ -0,0 +1,151 @@
+        fi
+        pr=$1
+        if ! [[ $2 =~ ^[0-9]+$ ]]; then
+            echo "ERROR: $2 is not a valid Chain ID number" 1>&2
+            usage
+            exit 1
+        fi
+        chainid="$(cast chain-id --rpc-url $3)"
+        if [[ $chainid != $2 ]]; then
+            echo "ERROR: RPC $3 doesn't match chain ID $2" 1>&2
+            usage
+            exit 1
+        fi
+        rpc=$3
+        if [[ ! " ${availableVersion[@]} " =~ " ${4} " ]]; then
+            echo "ERROR: $4 is not a valid version" 1>&2
+            usage
+            exit 1
+        fi
+        version=$4
+        ;;
+    *)
+        usage
+        exit 1
+        ;;
+esac
+
+echo "Verifying Deployment Asset"
+gh pr diff $pr --patch | git apply --include 'src/assets/**'
+
+# Getting default addresses
+oneThreeZeroFiles=("compatibility_fallback_handler" "create_call" "gnosis_safe_l2" "gnosis_safe" "multi_send_call_only" "multi_send" "proxy_factory" "sign_message_lib" "simulate_tx_accessor")
+oneFourOneFiles=("compatibility_fallback_handler" "create_call" "multi_send_call_only" "multi_send" "safe_l2" "safe_proxy_factory" "safe" "sign_message_lib" "simulate_tx_accessor")
+versionFiles=()
+versionAddresses=()
+
+if [[ $version == "1.3.0" ]]; then
+    versionFiles=("${oneThreeZeroFiles[@]}")
+elif [[ $version == "1.4.1" ]]; then
+    versionFiles=("${oneFourOneFiles[@]}")
+fi
+
+# Getting default addresses
+for file in "${versionFiles[@]}"; do
+    versionAddresses+=($(jq -r '.defaultAddress' "src/assets/v$version/$file.json"))
+done
+
+# Getting default address based on the chain id
+versionAddressesCI=()
+for file in "${versionFiles[@]}"; do
+    versionAddressesCI+=($(jq -r '.networkAddresses["'$chainid'"]' "src/assets/v$version/$file.json"))
+done
+
+# Checking default addresses and the addresses created for that chain id
+for i in {0..8}; do
+    if [[ ${versionAddresses[$i]} != ${versionAddressesCI[$i]} ]]; then
+        echo "ERROR: "${versionFiles[$i]}" default address is not the same as the one created for the chain id" 1>&2
+        exit 1
+    fi
+done
+
+echo "Default addresses are correct"
+
+# Code Hash of the default addresses (Chain ID: 1)
+# cast keccak $(cast code contractAddress --rpc-url https://rpc.ankr.com/eth)
+
+oneThreeZeroContracts=("CompatibilityFallbackHandler" "CreateCall" "GnosisSafeL2" "GnosisSafe" "MultiSendCallOnly" "MultiSend" "ProxyFactory" "SignMessageLib" "SimulateTxAccessor")
+oneFourOneContracts=("CompatibilityFallbackHandler" "CreateCall" "MultiSendCallOnly" "MultiSend" "SafeL2" "SafeProxyFactory" "Safe" "SignMessageLib" "SimulateTxAccessor")
+versionContracts=()
+if [[ $version == "1.3.0" ]]; then
+    versionContracts=("${oneThreeZeroContracts[@]}")
+elif [[ $version == "1.4.1" ]]; then
+    versionContracts=("${oneFourOneContracts[@]}")
+fi
+
+versionCodeHashes=()
+for contract in "${versionContracts[@]}"; do
+    versionCodeHashes+=($(jq -r '.["'$version'"].["'$contract'"]' src/assets/code-hashes.json))
```

Comment:
One disadvantage of doing it this way over using `--arg` like I suggested is that special characters might break the `jq` query this way.

Reply:
Updated 👍🏾